### PR TITLE
executor: support Chunk for ShowExec

### PR DESCRIPTION
**Pull request by zz-jason**
to #5261

### CODE REVIEW

File: executor/show.go [hunk 1]
```diff
@@ -88,72 +92,100 @@ func (e *ShowExec) Next(goCtx goctx.Context) (Row, error) {
...
-		"DEFAULT",
-		"Supports transactions, row-level locking, and foreign keys",
-		"YES",
-		"YES",
-		"YES",
-	)
-	e.rows = append(e.rows, row)
+func (e *ShowExec) fetchShowEngines(forChunk bool) error {
+	if forChunk {
+		e.result.AppendString(0, "InnoDB")
```
Comment (winkyao) [inline executor/show.go]: use col :=0, and `e.result.AppendString(col, "InnoDB")` then col++ ?

Comment (winkyao) [inline executor/show.go]:
```
col := 0
e.result.AppendString(col, "InnoDB")
col++
```
maybe better?

File: executor/show.go [hunk 2]
```diff
@@ -88,72 +92,100 @@ func (e *ShowExec) Next(goCtx goctx.Context) (Row, error) {
...
+	chk.Append(e.result, e.cursor, e.cursor+numCurBatch)
+	e.cursor += numCurBatch
+	return nil
+}
+
+func (e *ShowExec) fetchAll(forChunk bool) error {
 	switch e.Tp {
 	case ast.ShowCharset:
-		return e.fetchShowCharset()
+		return e.fetchShowCharset(forChunk)
```
Comment (shenli) [inline executor/show.go]: Why not add a attribute in ShowExec named forChunk?

File: executor/show.go (hunk 1 above)
Comment (zz-jason) [reply executor/show.go]: It's not straight forward and brief

File: executor/show.go [hunk 3]
```diff
@@ -141,15 +167,24 @@ func (e *ShowExec) fetchAll() error {
...
-		"YES",
-	)
-	e.rows = append(e.rows, row)
+	if e.forChunk {
+		e.result.AppendString(0, "InnoDB")
+		e.result.AppendString(1, "DEFAULT")
+		e.result.AppendString(2, "Supports transactions, row-level locking, and foreign keys")
+		e.result.AppendString(3, "YES")
+		e.result.AppendString(4, "YES")
+		e.result.AppendString(5, "YES")
```
Comment (lamxTyler) [inline executor/show.go]: Can we implement something like `MakeDatums` so that we could save lots of repeated work?

File: executor/show.go [hunk 4]
```diff
@@ -220,13 +245,11 @@ func (e *ShowExec) fetchShowTables() error {
 	}
 	sort.Strings(tableNames)
 	for _, v := range tableNames {
-		data := types.MakeDatums(v)
 		if e.Full {
-			// TODO: support "VIEW" later if we have supported view feature.
```
Comment (lamxTyler) [inline executor/show.go]: Keep the comment.

**Comment by winkyao** [comment]
plz fix ci.

**Pull request merged**